hooks/tk-multi-publish2/maya/publish_files_model.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import maya.cmds as cmds
+import maya.mel as mel
+import sgtk
+from sgtk.util.filesystem import ensure_folder_exists
+
+# DD imports
+import dd.runtime.api
+dd.runtime.api.load('wam')
+from wam.core import Workflow
+from wam.datatypes.element import Element
+
+dd.runtime.api.load('modelpublish')
+from modelpublish.lib.introspection import findModelRootNodes
+
+# for validate workflow
+dd.runtime.api.load('indiapipeline')
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+# TODO: move this to a config?
+validation_action_ret_value = {
+    "MultipleShapes": False,
+    "DrawingOverride": False,
+    "DuplicateNames": False,
+    "CheckAssetName": False,
+    "CheckGeoNames": False,
+    "CleanNamespace": False,
+    "ReferenceCheck": False,
+    "DeleteColorSets": False,
+    "CleanUpScene": False,
+}
+
+class MayaPublishFilesModelPlugin(HookBaseClass):
+    """
+    Inherits from MayaPublishFilesPlugin
+    """
+    def validate(self, task_settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+        Additional check for maya scene using validations from modelpublish.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        if item.type == "file.maya":
```

Comment:
Just as a heads up, when you port this to 3.x, you'll want to change this to `if item.type == "maya.session"`.

Reply:
Thanks!